File: fm-domain/src/main/java/io/zeitwert/fm/app/adapter/api/rest/dto/HomeActionResponse.java
package io.zeitwert.fm.app.adapter.api.rest.dto;

import io.zeitwert.dddrive.ddd.adapter.api.jsonapi.dto.EnumeratedDto;
import lombok.Builder;
import lombok.Data;

import java.time.OffsetDateTime;

@Data
@Builder
public class HomeActionResponse {

	private EnumeratedDto item;
	private Integer seqNr;
	private OffsetDateTime timestamp;
	private EnumeratedDto user;
	private String changes;
	private EnumeratedDto oldCaseStage;
	private EnumeratedDto newCaseStage;

}

File: fm-domain/src/main/java/io/zeitwert/fm/app/adapter/api/rest/dto/HomeActionResponse.kt
package io.zeitwert.fm.app.adapter.api.rest.dto

import io.zeitwert.dddrive.ddd.adapter.api.jsonapi.dto.EnumeratedDto
import java.time.OffsetDateTime

data class HomeActionResponse(
	val item: EnumeratedDto,
	val seqNr: Int,
	val timestamp: OffsetDateTime,
	val user: EnumeratedDto,
	val changes: String? = null,
	val oldCaseStage: EnumeratedDto? = null,
	val newCaseStage: EnumeratedDto,
)
